gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -528,10 +528,9 @@ private int readObjectContentFromGCS(ByteBuffer byteBuffer) throws IOException {
       int skipBytes = Math.toIntExact(positionForNextRead - positionInGrpcStream);
       if (skipBytes >= 0 && skipBytes < content.size()) {
         content = content.substring(skipBytes);
-        positionInGrpcStream = positionForNextRead;
+        positionInGrpcStream += skipBytes;
       } else if (skipBytes >= content.size()) {
         positionInGrpcStream += content.size();
-        positionForNextRead = positionInGrpcStream;
         return bytesRead;
```

Comment:
this will always be 0, correct? Can we change this to 0 and move byteRead variable closer to where it is getting used?

Reply:
Will refactor this code.